modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientSchema.java

```diff
@@ -39,6 +65,15 @@
     /** Columns map by name. */
     private final Map<String, ClientColumn> map = new HashMap<>();
 
+    /** Marshaller cache. */
+    private final Map<Mapper, Marshaller> marshallers = new ConcurrentHashMap<>();
+
+    /** Marshaller cache. */
+    private final Map<Mapper, Marshaller> keyMarshallers = new ConcurrentHashMap<>();
+
+    /** Marshaller cache. */
+    private final Map<Mapper, Marshaller> valMarshallers = new ConcurrentHashMap<>();
```

Comment:
What about cache invalidation?
Mapper doesn't implement hashcode/equals., so using mapper builder will bloat the cache.

Reply:
Removed, added TODO with https://issues.apache.org/jira/browse/IGNITE-16094 link.